services/EventTokenService.js

```diff
@@ -0,0 +1,68 @@
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+
+// A service class for Token generation and management
+
+class EventTokenService {
+  static #app_access_key = process.env.LIVE_APP_ACCESS_KEY || config.get("Event100ms.APP_ACCESS_KEY");
+  static #app_secret = process.env.LIVE_APP_SECRET || config.get("Event100ms.APP_SECRET");
+  #managementToken;
+  constructor() {
+    this.#managementToken = this.getManagementToken(true);
+  }
+
+  // A private method that uses JWT to sign the payload with APP_SECRET
+  #signPayloadToToken(payload) {
```

Comment:
in the line return !exp || exp + buffer < currTimeSeconds;. This line is checking to see if the token has expired. However, if the exp value is null or undefined, then the || operator will always evaluate to true, even if exp + buffer < currTimeSeconds is false. This means that the function will always return true, even if the token is not expired.

Reply:
As discussed, we don't need to check for the exp while returning. Thank you for the comment. I have removed the check.